v-next/hardhat/src/internal/cli/main.ts

```diff
@@ -45,18 +46,26 @@ export async function main(
   cliArguments: string[],
   print: (message: string) => void = console.log,
 ): Promise<void> {
+  const log = debug("hardhat:core:cli:main");
+
   let builtinGlobalOptions;
 
+  log("Hardhat CLI started with arguments:", cliArguments);
+
   try {
     const usedCliArguments: boolean[] = new Array(cliArguments.length).fill(
       false,
     );
 
+    log("Parsing builtin global options");
```

Comment:
I think printing just the results would probably be enough in most cases, and make the output less verbose/more helpful.

Reply:
I mean in general. I noticed a pattern of "doing this", "this is its result", which I do a ton for manual debugging, but can be too verbose.